### DISCUSSION TITLE
106. 복합 칼럼 인덱스를 효율적으로 설계하기 위해 고려해야 할 요소는 무엇인가요?

### DISCUSSION THREAD

**Discussion by jongmee** [jewelry-box/cs-interview#110] [General]
키워드: 복합 칼럼 인덱스, 다중 칼럼 인덱스, 인덱스 설계, 선택도(카디널리티)

**Comment by cutehumanS2** [comment]
복합 칼럼 인덱스를 효율적으로 설계하기 위해 고려해야 할 요소는 **칼럼 순서**와 **칼럼의 선택도**입니다. 

- 복합 칼럼 인덱스는 첫 번째 칼럼부터 순차적으로 정렬되므로, where 조건에서 선행 칼럼이 사용되지 않으면 인덱스 활용이 제한될 수 있습니다.
- 선택도가 높은(=중복이 적은) 칼럼을 앞에 배치하면 필터링 범위가 줄어들어 성능이 향상됩니다.

<hr>

**[+) 선택도 관련 질문 살펴보기](https://github.com/jewelry-box/cs-interview/discussions/81)**

<br>

[참고한 자료]
책 <Real MySQL 8.0 (1권)> - 8.3.3.3 선택도(기수성)
